iamctl/pkg/userStores/export.go

```diff
@@ -75,66 +79,38 @@ func exportUserStore(userStoreId string, outputDirPath string, format string) er
 		fileType = utils.MEDIA_TYPE_YAML
 	}
 
-	var reqUrl = utils.SERVER_CONFIGS.ServerUrl + "/t/" + utils.SERVER_CONFIGS.TenantDomain + "/api/server/v1/userstores/" + userStoreId + "/export"
-
-	var err error
-	req, err := http.NewRequest("GET", reqUrl, strings.NewReader(""))
+	resp, err := utils.SendExportRequest(userStoreId, fileType, utils.USERSTORES, true)
 	if err != nil {
-		return fmt.Errorf("error while creating the request to export user store: %s", err)
-	}
-	req.Header.Set("Content-Type", utils.MEDIA_TYPE_FORM)
-	req.Header.Set("accept", fileType)
-	req.Header.Set("Authorization", "Bearer "+utils.SERVER_CONFIGS.Token)
-
-	defer req.Body.Close()
-
-	httpClient := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		},
+		return fmt.Errorf("error while exporting the identity provider: %s", err)
 	}
 
-	resp, err := httpClient.Do(req)
+	var attachmentDetail = resp.Header.Get("Content-Disposition")
+	_, params, err := mime.ParseMediaType(attachmentDetail)
 	if err != nil {
-		return fmt.Errorf("error while exporting the user store: %s", err)
+		return fmt.Errorf("error while parsing the content disposition header: %s", err)
 	}
-	defer resp.Body.Close()
-
-	statusCode := resp.StatusCode
-	if statusCode == 200 {
-		var attachmentDetail = resp.Header.Get("Content-Disposition")
-		_, params, err := mime.ParseMediaType(attachmentDetail)
-		if err != nil {
-			return fmt.Errorf("error while parsing the content disposition header: %s", err)
-		}
 
-		fileName := params["filename"]
-		exportedFileName := filepath.Join(outputDirPath, fileName)
-		fileInfo := utils.GetFileInfo(exportedFileName)
+	fileName := params["filename"]
+	exportedFileName := filepath.Join(outputDirPath, fileName)
+	fileInfo := utils.GetFileInfo(exportedFileName)
```

Comment:
are the possible exceptions here handled through the utils method?

Reply:
This method simply extracts the basic details about a file such as the resource name, file extension, etc of the file. There is no possibility of getting exceptions here.